src/main/java/org/truffleruby/core/fiber/FiberManager.java

```diff
@@ -61,6 +61,15 @@ public DynamicObject getRootFiber() {
     }
 
     public DynamicObject getCurrentFiber() {
+        assert Layouts.THREAD.getFiberManager(context.getThreadManager().getCurrentThread()) == this :
+            "FiberManager#getCurrentFiber() must be called on a Fiber belonging to the FiberManager";
```

Comment:
I'm confused by this message. This method can never be called on a `Fiber`, since it's an instance method on `FiberManager`.

Reply:
It's difficult to express.
One idea: `The current Fiber can only be read by the Thread the Fiber belongs to, otherwise it's racy`.

Reply:
Or maybe `Trying to read the current Fiber of another Thread which is inherently racy`.

Reply:
Too late for this merge, but please don't require that Fibers can only be resumed by the Thread that created them. That's an implementation detail in MRI and other runtimes should not limit themselves. I know it's racy, but so are lots of things. See my ticket: https://bugs.ruby-lang.org/issues/13821

Reply:
@chuckremes I replied on the ticket.
In TruffleRuby we have the same restriction as MRI currently.
It simplifies the implementation quite a bit and maintains the guaranty that two fibers of the same Thread never execute concurrently and only pass control to each other explicitly, not by another thread stealing a fiber.